department-of-veterans-affairs/vets-website
# Pension 71730 mvp add medical expenses

## Summary
On the Pension Benefits application(21P-527EZ), update the 'Financial information' chapter to collect information about unreimbursed expenses.

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Start or continue the application
3. Go to the 'Financial Information' chapter
4. Continue to the 'Medical expenses' page
5. Verify that the 'Add a medical or other unreimbursed expense' page is the next page when 'Yes' is selected
6. Verify that the 'Add a medical or other unreimbursed expense' page is NOT the next page when 'No' is selected
7. Verify the layout and content on the 'Add a medical or other unreimbursed expense' page
8. Verify that appropriate fields are required to add an unreimbursed expense
9. Verify that a valid entered medical expanse collapses when [ Add another unreimbursed expense ] is pressed
10. Verify that you can edit and add multiple unreimbursed medical expenses 
12. Verify that the form data is saved to the redux store and submitted to the api

## Related issue(s)
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/71730)
[Epic in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/63349)

## Testing done

## Screenshots

### Desktop
| Add unreimbursed medical expense | Valid | Invalid | Multiple medical expenses |
| ------ | ------ | ------ | ------ |
|![localhost_3001_pension_application_527EZ_financial_medical-expenses_add](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/20dcd6e9-f3af-4bd0-baad-c96a8c8148fc)|![localhost_3001_pension_application_527EZ_financial_](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/bf2bc64e-0287-4310-b0a3-6d3a1d0da965)|![localhost_3001_pension_application_527EZ_financial_ (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/5b580ffd-eaa9-4fdc-9c09-425f57db794b)|![localhost_3001_pension_application_527EZ_ (1)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/848260a7-aede-4803-b7b0-adb42b8c9e6d)

## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
[Ticket in Github](https://github.com/department-of-veterans-affairs/va.gov-team/issues/71730)

### Quality Assurance & Testing

### Error Handling


### Authentication
